pip/commands/list.py

```diff
@@ -169,6 +181,15 @@ def get_uptodate(self, packages, options):
             if dist.latest_version == dist.parsed_version
         ]
 
+    def get_nodeps(self, packages, options):
+        installed_packages = [
+            dist for dist in self.iter_packages_latest_infos(packages, options)]
+        all_dependencies = []
+        for dist in installed_packages:
+            all_dependencies.extend(dist.requires())
+        dep_keys = set(dep.key for dep in all_dependencies)
+        return set(pkg for pkg in installed_packages if pkg.key not in dep_keys)
```

Comment:
nit: you don't need a temporary list `all_dependencies`, I think you could do:

    dependencies_keys = set()
    for dist in installed_packages:
        dependencies_keys.update(dep.key for dep in dist.requires())
    return set(pkg for pkg in installed_packages if pkg.key not in dependencies_keys)